pytorch_toolkit/ote/ote/misc.py

```diff
@@ -23,3 +32,83 @@ def get_file_size_and_sha256(snapshot):
         'name': os.path.basename(snapshot),
         'source': snapshot
     }
+
+
+class NonBlockingStreamReader:
+
+    def __init__(self, stream):
+        self.stream = stream
+        self.queue = Queue()
+
+        def populate_queue(stream, queue):
+            while True:
+                line = stream.readline()
+                if line:
+                    queue.put(line)
+                else:
+                    time.sleep(1)
+
+        self.thread = Thread(target=populate_queue, args=(self.stream, self.queue))
+        self.thread.daemon = True
+        self.thread.start()
+
+    def readline(self, timeout=None):
+        try:
+            return self.queue.get(block=timeout is not None, timeout=timeout)
+        except Empty:
+            return None
+
+
+def get_complexity_and_size(cfg, config_path, work_dir, update_config):
+    """ Gets complexity and size of a model. """
+
+    image_shape = [x['img_scale'] for x in cfg.test_pipeline if 'img_scale' in x][0][::-1]
+    image_shape = " ".join([str(x) for x in image_shape])
```

Comment:
Unfortunately we may have different definitions of `image_shape`: 2D for detectors and 3D for action models. It looks like the shape definition should be moved to function parameters.

Reply:
Added possibility to specify image shape in complexity evaluation. Is it OK?

Reply:
It looks not well (assumes that 2D networks are ordinary case and the rest is not necessary) but the future generalization of code is unclear. So OK.